tests/unit_tests/test_ustar_cp/test_fcEqnAnnualSine.py

```diff
@@ -0,0 +1,59 @@
+import pytest
+import matlab.engine
+import numpy as np
+
+@pytest.mark.parametrize(
+    "b, t, expected",
+    [
+        # Case 1: Zero Amplitude (Constant)
+        ([0, 0, 0], [0, 90, 180, 270, 360], [0, 0, 0, 0, 0]),
+
+        # Case 2: Pure Sine Wave, Zero Offset
+        ([0, 1, 0], [0, 90, 180, 270, 360], [0, 0.9997458409224182, 0.04507749933823567, -0.9977133433981337, -0.09006335547871482]),
+
+        # Case 3: Sine Wave with Offset
+        ([2, 1, 0], [0, 90, 180, 270, 360], [2, 2.9997458409224182, 2.0450774993382357, 1.0022866566018664, 1.909936644521285]),
+
+        # Case 4: Phase Shift
+        ([0, 1, 90], [0, 90, 180, 270, 360], [-0.9997458409224182, 0, 0.9997458409224182, 0.04507749933823567, -0.9977133433981337]),
+
+        # Case 5: Negative Amplitude
+        ([0, -1, 0], [0, 90, 180, 270, 360], [0, -0.9997458409224182, -0.04507749933823567, 0.9977133433981337, 0.09006335547871482]),
+
```

Comment:
Could we add a little variation, e.g., with smaller time frames and non monotonic time frames?
```suggestion
        # Case 6: Pure Sine Wave, Zero Offset, small time frame
        ([0, 1, 0], [0, 180, 360], [0, 0.04507749933823567, -0.09006335547871482]),

        # Case 7: Pure Sine Wave, Zero Offset, non monotonic time frame
        ([0, 1, 0], [0, 360, 180], [0,  -0.09006335547871482, 0.04507749933823567]),
```